CRM/Extension/ClassLoader.php

```diff
@@ -164,6 +164,9 @@ private static function loadExtension(\Composer\Autoload\ClassLoader $loader, CR
 
           case 'psr4':
             $loader->addPsr4($mapping['prefix'], $path . '/' . $mapping['path']);
+            if (defined('CIVICRM_TEST')) {
+              $loader->addPsr4($mapping['prefix'], $path . '/tests/phpunit/' . $mapping['path']);
```

Comment:
@demeritcowboy just wondering if we should be testing for the presence of the test folder as well but probably doesn't matter that much

Reply:
Interesting question - yeah probably. Composer does [this](https://github.com/composer/composer/blob/39de9899a70d8351db134027dc24ed35fc629346/src/Composer/Autoload/ClassLoader.php#L521), so if the array is smaller to begin with it should be more efficient, trading one is_dir call here for several file_exists later. And then it would make sense to do it for non-test folders too.

Let's add it. For unscientific testing purposes the first one "Took 2 hr 31 min on test-1".